### ISSUE TITLE
Migrate from reStructuredText to Markdown

### ISSUE THREAD

**Issue by blackwind** [EventGhost/EventGhost#18]
With `sphinx` [now capable of building Markdown documents](https://github.com/sphinx-doc/sphinx/issues/825#issuecomment-180197293), with Markdown being the more natural and easy-to-remember of the two, and with Markdown being the format of choice here on GitHub and in the developer community at large, we should discuss migrating from reStructuredText to Markdown. Doing so will allow us to display the exact same changelog (see #15) inside EventGhost's about dialog and here on GitHub.

Things to keep in mind:

* Plugin descriptions (displayed in the plugin settings dialog) should be updated to Markdown and denoted with "\<md\>" at the start, as opposed to "\<rst\>" as it is now.
* A `DecodeMarkdown()` function should be added to `eg.Utils`.
* `eg.Utils.DecodeReST()` should remain in place for plugins not under our control that we can't migrate to Markdown ourselves.

Can anyone think of any reason not to migrate?

**Comment by blackwind** [comment]
Pushing back the milestone on this one due to the [current limitations](https://blog.readthedocs.com/adding-markdown-support/#limitations) of CommonMark.

**Comment by blackwind** [comment]
The foundation for this is now complete as of #59. All that remains is:

* Convert plugin descriptions -- possible now, but I'm in no rush.
* Convert documentation -- still impossible at this time.

**Comment by WoLpH** [comment]
I would vote against switching as long as the markdown parser is a 3rd party extension. The choice for reStructuredText within Sphinx is due to the extensibility. Extensions such as InterSphinx will most likely not function when using markdown.

**Comment by dequi** [comment]
While testing converting plugin descriptions:
```
eg.RegisterPlugin(
    description = <accepts markdown and ReST>,
    help = <does not accept markdown, nor ReST>,
)
```
Any intentions on getting GitHub Flavored markdown up and running? I'd love to be able to just import a `README.md` file as an argument in the `help` parameter.

**Comment by WoLpH** [comment]
Actually, Github supports README.rst as well so you can probably do that already.

**Comment by dequi** [comment]
No, tried going down that path. The `help` parameter doesn't accept ReSTRUCTURED either. Don't want to use the `description` for that. `description` should be kept short and easy maybe to be read out by a script.  Well that and this issue is about migrate to markdown :smiley:
